components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 LLC. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content.
+ *
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * This class is model class of a federated token.
+ * A federated token is an external token obtained via an OIDC federated authenticator
+ * after a successful authentication.
+ */
+public class FederatedToken implements Serializable {
+
+    private static final long serialVersionUID = 6618332057931299623L;
+    private String idp;
+    @JsonProperty("expires_in")
```

Comment:
Lets not use this class to directly build the response in the oauth level

Reply:
Fixed cfb085b712df445a0a612df800c012b16fa75faf